### PR TITLE
feat(docs): Improve documentation about Search

### PR DESCRIPTION
**Pull request by szalai1**
(cherry picked from commit ec016b4abff6bdc5ae753f14cca4746944b7ed82)

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by jjoyce0510 — APPROVED** [review]
I made various changes directly as commits on this PR. Overall looking good! 

Once @gabe-lyons 's advanced search features land, we'll need to come back and revisit this

**Comment by maggiehays** [comment]
@szalai1 @jjoyce0510 we get a lot of questions from OSS about how Search results are displayed - can we add in those details in FAQ?

File: docs/how/search.md [hunk 1]
```diff
@@ -1,17 +1,76 @@
...
-The following examples are in the format of  
-X: *typical question* :  
+Add in `saasOnly` for Managed DataHub-only features
+ -->
+
+<FeatureAvailability/>
+
+The **search bar** is an important mechnaism for discovering data assets in DataHub. From the search bar, you can find Datasets, Columns, Dashboards, Charts, Data Pipelines, and more. Simply type in a term and press 'enter'. 
+
+![](https://github.com/datahub-project/static-assets/blob/main/imgs/search-landingpage.png?raw=true)
```
Comment (maggiehays) [inline docs/how/search.md:17]: Small nit-pick for consistency with other guides & improved readability - please use HTML to embed images so we can size down (70%) & center-align: 

```
<p align="center">
  <img width="70%"  src="https://github.com/datahub-project/static-assets/blob/main/imgs/search-landingpage.png?raw=true"/>
</p>
```

File: docs/how/search.md [hunk 2]
```diff
@@ -1,17 +1,76 @@
...
+* Where/how do you access it?
+* What best practices exist?
+* What are common code snippets?
+ -->
+ 
+### Filters
+
+The filters sidebar sits on the left hand side of search results, and lets users find assets by drilling down. You can quickly filter by Data Platform (e.g. Snowflake), Tags, Glossary Terms, Domain, Owners, and more with a single click. 
+
+![](https://raw.githubusercontent.com/datahub-project/static-assets/main/imgs/filters_highlighted.png)
```
Comment (maggiehays) [inline docs/how/search.md:56]: Same here - please use HTML to embed:
```
<p align="center">
  <img width="70%"  src="https://raw.githubusercontent.com/datahub-project/static-assets/main/imgs/filters_highlighted.png"/>
</p>
```

File: docs/how/search.md [hunk 3]
```diff
@@ -1,17 +1,76 @@
...
+* In which DataHub Version did this become available? -->
+
+## Search Setup, Prerequisites, and Permissions
+
+<!-- This section should provide plain-language instructions on how to configure the feature:
+
+* What special configuration is required, if any?
+* How can you confirm you configured it correctly? What is the expected behavior?
+* What access levels/permissions are required within DataHub? -->
+Search is available for all users. Although Search works out of the box, the more relevant data you ingest, the better the results are.
```
Comment (maggiehays) [inline docs/how/search.md:35]: Let's elaborate on this - what is relevant data here? What types of metadata improve Search results?

File: docs/how/search.md [hunk 4]
```diff
@@ -1,17 +1,76 @@
...
+
+<FeatureAvailability/>
+
+The **search bar** is an important mechnaism for discovering data assets in DataHub. From the search bar, you can find Datasets, Columns, Dashboards, Charts, Data Pipelines, and more. Simply type in a term and press 'enter'. 
+
+![](https://github.com/datahub-project/static-assets/blob/main/imgs/search-landingpage.png?raw=true)
+
+**Advanced queries** and the **filter sidebar** helps fine tuning queries. For programmatic users Datahub provides a **GraphQL API** as well. 
+
+<!-- This section should provide a plain-language overview of feature. Consider the following:
```
Comment (maggiehays) [inline docs/how/search.md:21]: Let's remove commented-out template details unless they are proving specific value in how to interpret the doc

**Review by maggiehays — CHANGES_REQUESTED** [review]
This is looking great!! Left a few comments/requested changes for you to consider

**Review by maggiehays — APPROVED** [review]
Look great!

**Comment by hsheth2** [comment]
The formatting of the use case list is pretty ugly - not super readable, extra `*` characters, some headers are italics and others not


image

**Pull request merged**